x-pack/docs/en/security/securing-communications/tls-versions-jdk.asciidoc

```diff
@@ -0,0 +1,144 @@
+of the Java installation.
+If `$JAVA_HOME` points to the home directory of the JDK that you use to run {es},
+then the configuration file will be in:
+
+* `$JAVA_HOME/jre/lib/security/java.security`
+
+For **JDK11 or later**, the configuration file is within the `conf/security`
+directory of the Java installation.
+If `$JAVA_HOME` points to the home directory of the JDK that you use to run
+{es}, then the configuration file will be in:
+
+* `$JAVA_HOME/conf/security/java.security`
+
+===== Copy the disabledAlgorithms setting
+
+Within the JDK configuration file is a line that starts with
+`jdk.tls.disabledAlgorithms=`.
+This setting controls which protocols and algorithms are _disabled_ in your JDK.
+The value of that setting will typically span multiple lines.
+
+For example, in OpenJDK 16 the setting is:
+[source,text]
+--------------------------------------------------
+jdk.tls.disabledAlgorithms=SSLv3, TLSv1, TLSv1.1, RC4, DES, MD5withRSA, \
+    DH keySize < 1024, EC keySize < 224, 3DES_EDE_CBC, anon, NULL
+--------------------------------------------------
+
+Create a new file in your in your {es} configuration directory named `es.java.security`.
+Copy the `jdk.tls.disabledAlgorithms` setting from the JDK's default configuration file into `es.java.security`.
+You do not need to copy any other settings.
+
+===== Enable required TLS versions
+
+Edit the `es.java.security` file in your {es} configuration directory, and
+modify the `jdk.tls.disabledAlgorithms` setting so that any SSL or TLS versions
+that you wish to use are no longer listed.
+
+For example, if to enable TLSv1.1 on OpenJDK (which uses the
+`jdk.tls.disabledAlgorithms` settings shown above), then the contents of the
+`es.java.security` file would include supported protocols _except_ `TLSv1.1`:
+
+[source,text]
+--------------------------------------------------
+jdk.tls.disabledAlgorithms=SSLv3, TLSv1, RC4, DES, MD5withRSA, \
+    DH keySize < 1024, EC keySize < 224, 3DES_EDE_CBC, anon, NULL
+--------------------------------------------------
+
+===== Enable your custom security configuration
+
+To enable your custom security policy, add a file in the <<set-jvm-options, `jvm.options.d`>>
+directory within your {es} configuration directory.
+
+1. Locate the `jvm.options.d` directory for your {es} installation.
+2. Create a file named `java.security.options` within the `jvm.options.d` directory.
+3. Set the path for the `es.java.security` file.
+
+[source,text]
+----
+-Djava.security.properties=/path/to/your/es.java.security
+----
+==== Enabling TLS versions in {es}
+
+SSL/TLS versions can be enabled and disabled within {es} via the
+<<ssl-tls-settings,`ssl.supported_protocols` settings>>.
+
+{es} will only support the TLS versions that are enabled by the
+<<jdk-tls-versions,underlying JDK>>. If you configure
+`ssl.supported_procotols` to include a TLS version that is not enabled in your
+JDK, then it will be silently ignored.
```

Comment:
Separately, I wonder whether it is possible to log some warnings for this situation.